core/libcamera_app.cpp

```diff
@@ -147,12 +151,25 @@ void LibcameraApp::ConfigureViewfinder()
 	// Now we get to override any of the default settings from the options_->
 	configuration_->at(0).pixelFormat = libcamera::formats::YUV420;
 	configuration_->at(0).size = size;
+
+	if (have_lores_stream)
+	{
+		Size lores_size(options_->lores_width, options_->lores_height);
+		lores_size.alignDownTo(2, 2);
+		if (lores_size.width > size.width || lores_size.height > size.height)
+			throw std::runtime_error("Low res image larger than viewfinder");
+		configuration_->at(1).pixelFormat = configuration_->at(0).pixelFormat;
```

Comment:
Do we need to be a bit careful here?  If configuration_->at(0).pixelFormat is an RGB format, the low res output will not be able to do that.  Although, I think the pipeline handler might correct this, not sure.

Reply:
Good point. I should hardcode the lo res stream always to YUV420.